docker-build.sh

```diff
@@ -141,17 +164,30 @@ do
   # Construct Dockerfile for build
 
   # start with upstream image
-  echo "FROM elasticsearch:${UPSTREAM_ES}" > Dockerfile
+  echo "FROM ${DOCKER_REPO}elasticsearch:${UPSTREAM_ES}" > Dockerfile
 
   # add all plugins
-  for plugin in ${plugins}
-  do
-     echo "RUN /usr/share/elasticsearch/bin/plugin install --batch ${plugin}" >> Dockerfile
-  done
+  if [ "${ver_maj}" -eq "1" ]
+  then
+     # hack to install migration plugin
+      echo "RUN /usr/share/elasticsearch/bin/plugin -i migration -u https://github.com/elastic/elasticsearch-migration/releases/download/v1.19/elasticsearch-migration-1.19.zip" >> Dockerfile
+  else   
+      for plugin in ${plugins}
+      do
+         case ${ver_maj} in
+         "2") # 2.x plugin insall
+             echo "RUN /usr/share/elasticsearch/bin/plugin install --batch ${plugin}" >> Dockerfile
+            ;;
+         *) # assume all newer uses the following commadn
```

Comment:
- Consider using a version-specific case statement section for all supported versions, instead of using a wildcard for all unspecified versions. This avoids unintended behavior if the plugin install method/syntax changes.
- If implementing the previous suggestion, change wildcard section to catch unsupported values of maj_ver, and exit with a return value > 1.
- Typo: "commadn" should be "command".